fusionauth/resource_fusionauth_reactor.go

```diff
@@ -0,0 +1,122 @@
+				Description: "The license Id to activate",
+			},
+			"license": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "The Base64 encoded license value. This value is necessary in an air gapped configuration where outbound network access is not available.",
+			},
+		},
+	}
+}
+
+func buildReactor(data *schema.ResourceData) fusionauth.ReactorRequest {
+	reactor := fusionauth.ReactorRequest{
+		LicenseId: data.Get("license_id").(string),
+		License:   data.Get("license").(string),
+	}
+	return reactor
+}
+
+func createReactor(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	reactor := buildReactor(data)
+
+	resp, faErrs, err := client.FAClient.ActivateReactor(reactor)
+	if err != nil {
+		return diag.Errorf("CreateReactor err: %v", err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Every terraform resource needs an id. Since none of the fusion auth reactor apis return
+	// the id of the reactor, we will need to make one up. There can only be one reactor
+	// so this should be safe.
+	data.SetId("fusion-auth-reactor-id")
+
+	return nil
+}
+
+func updateReactor(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+	reactor := buildReactor(data)
+
+	resp, faErrs, err := client.FAClient.ActivateReactor(fusionauth.ReactorRequest{
+		LicenseId: reactor.LicenseId,
+		License:   reactor.License,
+	})
+	if err != nil {
+		return diag.Errorf("UpdateReactor err: %v", err)
+	}
+
+	if err := checkResponse(resp.StatusCode, faErrs); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func deleteReactor(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+
+	resp, err := client.FAClient.DeactivateReactor()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, nil); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func readReactor(_ context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client := i.(Client)
+
+	resp, err := client.FAClient.RetrieveReactorStatus()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := checkResponse(resp.StatusCode, nil); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// The reactor read api does not return the license_id or license values. So we
+	// cannot update the state using this response like the other resources.
+	// However, we know that if the "licensed" attribute is false we can unset those attributes.
+	// TODO: how will the state get updated after create or update?
+	if !resp.Status.Licensed {
+		if err := data.Set("license_id", ""); err != nil {
+			return diag.Errorf("reactor.license_id: %s", err.Error())
+		}
+		if err := data.Set("license", ""); err != nil {
+			return diag.Errorf("reactor.license: %s", err.Error())
+		}
+	}
```

Comment:
This is my biggest unknown. I *think* by just not setting the `data` parameter at all here when the reactor is licensed, then the state will just get updated with the values from the configuration.